pkg/codegen/pcl/binder_resource.go

```diff
@@ -497,6 +497,19 @@ func bindResourceOptions(options *model.Block) (*ResourceOptions, hcl.Diagnostic
 	return resourceOptions, diagnostics
 }
 
+// unwrapOptionalType returns T from an optional type that is modelled as Union[T, None] or Union[None, T]
+// if the input type is not optional, it returns the input type as is.
+func unwrapOptionalType(t model.Type) model.Type {
```

Comment:
I think it makes sense to let the user know this wasn't optional if it wasn't with an error return, isn't that go-y?

Reply:
It's not really an error if something is not optional in this case. We are changing the type here so `model.Type -> model.Type` makes sense. I only want to error if it is expected to error _and_ I want to act upon it which is not the case here